lib/collection/src/recommendations.rs

```diff
@@ -274,17 +311,17 @@ fn recommend_by_avg_vector(
         Some(UsingVector::Name(name)) => name,
     };
 
-    let avg_positive = avg_vectors(positive_vectors);
+    let avg_positive = avg_vectors(positive_vectors)?;
     let negative = negative_vectors.collect_vec();
 
     let search_vector = if negative.is_empty() {
         avg_positive
     } else {
-        let avg_negative = avg_vectors(negative.into_iter());
-        merge_positive_and_negative_avg(avg_positive, avg_negative)
+        let avg_negative = avg_vectors(negative.into_iter()).unwrap();
```

Comment:
why is this unwrap safe?

if the negative vectors are mixed this could panic?

Reply:
I meant what if the `negative` contains a mix of dense and sparse vectors.
It seems `avg_vectors` returns an error in this case.